src/ENVIV.cpp

```diff
@@ -0,0 +1,103 @@
+#ifdef SENSORS
+#include "ENVIV.h"
+#include "globals.h"
+#include "mqtt.h"
+#include "defaults.h"
+#include <AsyncWiFiSettings.h>
+#include "string_utils.h"
+
+#include <M5UnitENV.h>
+
+namespace ENVIV
+{
+    BMP280 bmp;
+    SHT4X sht;
+    //long BMP280_status;
+    String ENVIVBMP280_I2c;
+    int ENVIVBMP280_I2c_Bus;  
+    unsigned long BMP280PreviousMillis = 0;
+    int sensorInterval = 60000;
+    bool initialized = false;
+    bool initializedsht = false;
+    
+    void Setup() {
+        //Serial.println("starting env IV setup");
+        if (!I2C_Bus_1_Started && !I2C_Bus_2_Started) return;
+        //Serial.println("bus check complete");
+
+        if (!bmp.begin(&Wire, BMP280_I2C_ADDR, 2, 1, 400000U)) {
+            Serial.println("[ENVIV BMP280] Couldn't find a sensor, check your wiring and I2C address!");
+            while (1) delay(1);
+        } else {
+            initialized = true;
+        }
+
+        /* Default settings from datasheet. */
+        bmp.setSampling(BMP280::MODE_NORMAL,     /* Operating Mode. */
+                        BMP280::SAMPLING_X2,     /* Temp. oversampling */
+                        BMP280::SAMPLING_X16,    /* Pressure oversampling */
+                        BMP280::FILTER_X16,      /* Filtering. */
+                        BMP280::STANDBY_MS_500); /* Standby time. */
```

Comment:
Not sure about this suggestion.  All other examples of the BMPs from both adafruit and m5 (within ESPresense and other examples both vendors provide) use the forced mode.  If this change would help with efficiency, speed, or heat generation on the chip impacting readings, sure, this would be a good suggestion.  Not sure how or what to test, so if this is something the project would like to see I'm happy to discuss.